modules/home/generic/keyboard.nix

```diff
@@ -0,0 +1,48 @@
+{ config, pkgs, lib, ... }:
+
+let
+  cfg = config.mine.keyboard;
+
+  layouts = {
+    azerty = {
+      layout = "fr";
+      variant = "";
+    };
+    bepo = {
+      layout = "fr";
+      variant = "bepo";
+    };
+    colemak = {
+      layout = "us";
+      variant = "colemak";
+    };
+    qwerty = {
+      layout = "us";
+      variant = "";
+    };
+    qwerty_intl = {
+      layout = "us";
+      variant = "intl";
+    };
+  };
+in
+
+with lib;
+
+{
+  options = {
+    mine.keyboard = {
+      layouts = mkOption {
+        type = types.listOf (types.enum (attrNames layouts));
+        default = [ "qwerty" ];
+        example = [ "colemak" "qwerty" ];
+        description = "Layouts to set";
+      };
+    };
+  };
+
+  config = {
+    home.keyboard.layout = toString (map (n: layouts.n.layout) cfg.layouts);
```

Comment:
This should be a comma-separated list. At first, I could not find it in the docs, but following the code path I tracked it down to a call to `setxkbmap` [here](https://github.com/rycee/home-manager/blob/b6e1d826850fdcd79e067677ec0d5ec20fdf12b7/modules/xsession.nix#L85-L94).

From man `setxkbmap`:

>        -layout name
> Specifies the name of the layout used to determine the components which make up the keyboard description. The -layout option may only be used once. Multiple layouts can be specified as a comma-separated list.

>        -variant name
> Specifies which variant of the keyboard layout should be used to determine the components which make up the keyboard description. The -variant option may only be used once. Multiple variants can be specified as a comma-separated list and will be matched with the layouts specified with -layout.